src/secrets.rs

```diff
@@ -1,22 +1,143 @@
 //! Types for working with registry access secrets
 
+use serde::Deserialize;
+
+use base64;
+use dirs;
+use std::collections::HashMap;
+use std::env;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+use std::process::{Command, Stdio};
+use std::str;
+use tracing::debug;
+
 /// A method for authenticating to a registry
 pub enum RegistryAuth {
     /// Access the registry anonymously
     Anonymous,
+
     /// Access the registry using HTTP Basic authentication
     Basic(String, String),
+
+    /// Access the registry using credentials configured in the Docker config file
+    Default,
```

Comment:
Nit: Should probably rename `DefaultKeychain` to match what this method is called in other places like go